pkg/liqo-controller-manager/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -438,6 +467,15 @@ func getPeeringPhase(foreignCluster *discoveryv1alpha1.ForeignCluster,
 			return discoveryv1alpha1.PeeringConditionStatusDisconnecting, resourceRequestDeletingReason,
 				fmt.Sprintf(resourceRequestDeletingMessage, foreignCluster.Status.TenantNamespace.Local), nil
 		}
+		// Return "pending" if we sent the ResourceRequest but the foreign cluster did not yet create the ResourceOffer
+		if resourceOffer == nil {
+			return discoveryv1alpha1.PeeringConditionStatusPending, resourceRequestPendingReason,
+				fmt.Sprintf(resourceRequestPendingMessage, foreignCluster.Status.TenantNamespace.Local), nil
+		}
+		if resourceOffer.Status.Phase != sharingv1alpha1.ResourceOfferAccepted {
```

Comment:
can we check for the status.VirtualKubeletStatus to be equal to Created? It should be set after the kubelet creation, while the phase is set when it is creating the deployment

(if I remember correctly) tell me if I'm wrong

Reply:
Yes, fair point - that would make the detection more robust.